api-services/authentication/useAuth.tsx

```diff
@@ -0,0 +1,202 @@
+import type { Dispatch, PropsWithChildren, SetStateAction } from 'react'
+import { createContext, useContext, useEffect, useRef, useState } from 'react'
+import { z } from 'zod'
+import keycloakInstance, { KeycloakService } from '../util/keycloak'
+import { createPersonalOrganization } from '../mutations/users/organization_mutations'
+import { getAPIServiceConfig } from '../config/config'
+
+const IdTokenClaimsSchema = z.object({
+  sub: z.string().uuid(),
+  email: z.string().email(),
+  email_verified: z.boolean(),
+  name: z.string(),
+  preferred_username: z.string(),
+  given_name: z.string(),
+  family_name: z.string(),
+  organization: z.object({
+    id: z.string().uuid(),
+    name: z.string()
+  }).optional()
+})
+
+export const fakeToken = IdTokenClaimsSchema.default({
+  sub: '18159713-5d4e-4ad5-94ad-fbb6bb147984',
+  email: 'max.mustermann@helpwave.de',
+  email_verified: true,
+  name: 'Max Mustermann',
+  preferred_username: 'max.mustermann',
+  given_name: 'Max',
+  family_name: 'Mustermann',
+  organization: {
+    id: '3b25c6f5-4705-4074-9fc6-a50c28eba406',
+    name: 'helpwave fake'
+  }
+})
+
+const config = getAPIServiceConfig()
+
+const UserFromIdTokenClaims = IdTokenClaimsSchema.transform((obj) => ({
+  id: obj.sub,
+  email: obj.email,
+  name: obj.name,
+  nickname: obj.preferred_username,
+  avatarUrl: `https://source.boringavatars.com/marble/80/${obj.sub}`,
+  organization: obj.organization
+}))
+
+export type User = z.output<typeof UserFromIdTokenClaims>
+
```

Comment:
Make the user object the same as for the user type in the users API or call it differently here for example identity

Reply:
I would stick with user for now. In service we never really introduced the concept of `Identity`. Its always a user. We called a user with a valid session since the beginning User. We should discuss this separately.